packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -86,19 +106,36 @@ class DatePicker extends React.Component {
     };
 
     this.handleChange = this.handleChange.bind(this);
+    this.handleChangeRaw = this.handleChangeRaw.bind(this);
     this.handleOnSelect = this.handleOnSelect.bind(this);
     this.handleOnClickOutside = this.handleOnClickOutside.bind(this);
   }
 
-  handleOnSelect() {
-    this.requestFocus();
+  handleOnSelect(selectedDate, event) {
+    // onSelect should only be invoked when selecting a datr from the picker.
```

Comment:
Nit: `datr` I think was meant to be `date`

Reply:
Updated: https://github.com/cerner/terra-core/pull/768/commits/3763caa7628bc29b961a755d2d760b627a28bbfa